transmart-core-api/src/main/groovy/org/transmartproject/core/tree/TreeNode.groovy

```diff
@@ -0,0 +1,128 @@
+package org.transmartproject.core.tree
+
+import org.transmartproject.core.ontology.OntologyTerm
+import org.transmartproject.core.ontology.OntologyTermTag
+import org.transmartproject.core.ontology.OntologyTermType
+
+/**
+ * The ontology node class that represents nodes in the ontology forest.
+ * (There may be multiple root nodes, hence the name 'forest'.)
+ * Intermediate nodes contain a collection of child nodes.
+ * Leaf nodes represent a concept (either variable or value) or a study in TranSMART.
+ */
+interface TreeNode {
+
+    /**
+     * API version string used for marshalling the object.
+     * @return the API version string.
+     */
+    String getApiVersion()
+
+    void setApiVersion(String version)
+
+    /**
+     * The domain object on which this node is based.
+     */
+    OntologyTerm getDelegate()
+
+    /**
+     * Retrieves the parent node if it exists (i.e., when it is not a root node).
+     * @return the parent node.
+     */
+    TreeNode getParent()
+
+    void setParent(TreeNode parent)
+
+    /**
+     * Retrieves the child nodes.
+     * @return the child nodes.
+     */
+    List<TreeNode> getChildren()
+
+    /**
+     * Retrieves the tags associated with the node.
+     * @return the list of tags.
+     */
+    List<OntologyTermTag> getTags()
+
+    void setTags(List<OntologyTermTag> tags)
```

Comment:
We don't have code outside core-db modifying tree nodes. Shouldn't we make this interface contains getters only?

Reply:
It is used outside the boundaries of the TreeResource.

Reply:
@gijskant Where is it used then? I see usage in `TreeService` only which implements TreeResource. Anyway is there reason to keep the logic that sets the field outside the core?